sympy/functions/special/polynomials.py

```diff
@@ -1111,7 +1115,9 @@ def fdiff(self, argindex=2):
 
     def _eval_rewrite_as_polynomial(self, n, x):
         from sympy import Sum
-        # TODO: Should make sure n is in N_0
+        # Make sure n is in N_0
```

Comment:
following syntax above "n \in N_0" and then update test to use assumptions and the same for the change below (which I have not added a comment to).